workspaces/src/network/transaction.rs

```diff
@@ -0,0 +1,323 @@
+use std::convert::TryInto;
+
+use near_crypto::KeyType;
+use near_primitives::transaction::{
+    Action, AddKeyAction, CreateAccountAction, DeleteAccountAction, DeleteKeyAction,
+    DeployContractAction, FunctionCallAction, StakeAction, TransferAction,
+};
+use near_primitives::views::FinalExecutionOutcomeView;
+
+use crate::network::{CallExecution, CallExecutionDetails, Network, ViewResultDetails};
+use crate::rpc::client::{
+    send_batch_tx_and_retry, Client, DEFAULT_CALL_DEPOSIT, DEFAULT_CALL_FN_GAS,
+};
+use crate::types::{AccessKey, AccountId, Balance, Gas, InMemorySigner, PublicKey, SecretKey};
+use crate::worker::Worker;
+use crate::Account;
+
+/// A set of arguments we can provide to a transaction, containing
+/// the function name, arguments, the amount of gas to use and deposit.
+#[derive(Debug, Clone)]
+pub struct Function {
+    name: String,
```

Comment:
probably worth changing this to be a ref and have a lifetime now. This will be changed in the future to have one so no need to clone early to avoid

Reply:
hmm, I'm not sure I see why it would get changed in the future? Is there something that's getting changed somewhere that I'm missing? `Function` should be donating its internals later on anyways, so there shouldn't be a wasted clone/allocation here

Reply:
this is all just internal though, and I meant like we end up converting a `Function` over to a `FunctionCallAction` later on which would take an owned string in its current form. But potentially could be changed in the future too. Not too big of a deal but I'll change it over to the ref version